<launch>

  <!-- Connect the robot to a keyboard teleop controller -->
  <node name="teleop_twist_keyboard" pkg="teleop_twist_keyboard" type="teleop_twist_keyboard.py" output="screen" respawn="true"/>

  <!-- Run the map server -->
  <arg name="map_file" default="$(find me5413_world)/maps/map.yaml"/>
  <node name="map_server" pkg="map_server" type="map_server" args="$(arg map_file)" />
  
  <!-- Launch the AMCL -->
   <include file="$(find me5413_world)/launch/amcl.launch" />
 <!-- <include file="$(find me5413_world)/launch/ekf_template.launch" /> -->
   
  <!-- Launch Move Base -->
  <include file="$(find me5413_world)/launch/move_base.launch" />
  
   <!-- Add stereo_image_proc node for depth processing with correct topic remapping -->
  <node pkg="stereo_image_proc" type="stereo_image_proc" name="stereo_image_proc" output="screen" ns="/front">
    <remap from="left/image_raw" to="left/image_raw" />
    <remap from="right/image_raw" to="right/image_raw" />
    <remap from="left/camera_info" to="left/camera_info" />
    <remap from="right/camera_info" to="right/camera_info" />
  </node>
  
  
  <!-- Launch Rviz with our settings -->
  <node type="rviz" name="rviz" pkg="rviz" args="-d $(find me5413_world)/rviz/navigation.rviz" output="log" respawn="true"/>
  <node ns="me5413_world" pkg="me5413_world" type="goal_publisher_node" name="goal_publisher_node" output="screen" />
  <node pkg="me5413_world" type="improved_templatematch.py" name="template_matcher" output="screen" />
</launch>
